__std__/nix/toolchain/library/pkgs.nix

```diff
@@ -11,33 +11,17 @@
 
 let
 
-  haskell-nix-bootstrap = import inputs.haskell-nix {
-
-    pkgs = import inputs.nixpkgs.path {
-      system = inputs.nixpkgs.system;
-    };
-
-    sourcesOverride = {
-      hackage = inputs.hackage-nix;
```

Comment:
No! We absolutely definitely want to keep pinning `hackage.nix` separately from `haskell.nix`, it's far more convenient!

Reply:
If you bump your index-state you need to bump hackage.nix. That will likely force you to rebuild a bunch of haskell stuff, but _not_ GHC. Whereas bumping haskell.nix can often force you to rebuild GHC, so is far more disruptive. I think everyone should pin them separately.